arch/arm/core/mpu/arm_mpu.c

```diff
@@ -42,8 +42,8 @@ BUILD_ASSERT((DT_FOREACH_STATUS_OKAY_NODE_VARGS(
 	 (DT_REG_SIZE(node_id) >= CONFIG_CORTEX_M_NULL_POINTER_EXCEPTION_PAGE_SIZE)) ||
 
 #define DT_NULL_PAGE_DETECT_NODE_EXIST                                                             \
-	(DT_FOREACH_STATUS_OKAY_NODE_VARGS(NULL_PAGE_DETECT_NODE_FINDER, zephyr_memory_attr) false)
-
+	(DT_FOREACH_OKAY_VARGS_zephyr_memory_region(NULL_PAGE_DETECT_NODE_FINDER,                  \
```

Comment:
this is an implementation/intermediate function, technically not guaranteed to continue to exist as it is not an API defined thing, you should be using `DT_FOREACH_STATUS_OKAY_VARGS(zephyr_memory_region NULL_PAGE_DETECT_NODE_FINDER
```suggestion
	(DT_FOREACH_STATUS_OKAY_VARGS(zephyr_memory_region, NULL_PAGE_DETECT_NODE_FINDER, zephyr_memory_attr) false)
```